File: storage/storage.go
package storage

import (
	"context"
	"io"
)

type ReadSeekerCloser interface {
	io.Reader
	io.Seeker
	io.Closer
}

type StorageProvider interface {
	Upload(ctx context.Context, file io.Reader, filename string, contentType string) (string, error)
	Delete(ctx context.Context, fileURL string) error
	GetProviderName() string
}

File: python_backend/storage/storage.py
from abc import ABC, abstractmethod

class StorageProvider(ABC):
    @abstractmethod
    def upload(self, context, file, filename: str, content_type: str) -> str:
        pass

    @abstractmethod
    def delete(self, context, file_url: str):
        pass

    @abstractmethod
    def get_provider_name(self) -> str:
        pass
